### PR TITLE
refactor: paystring remove redux and payid-lib

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

Switch from redux to react-query and payid-lib to @paystring/utils.

This eliminates ~200 lines.

This is a recreation of #639.

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### TypeScript/Hooks Update

- [x] Updated files to React Hooks
- [x] Updated files to TypeScript

### CODE REVIEW

**Review by pdp2121 — APPROVED** [review]

File: src/containers/PayStrings/PayStringMappingsTable/test/PayStringMappingsTable.test.js [hunk 1]
```diff
@@ -1,86 +1,44 @@
...
 
   it('renders without crashing', () => {
-    const wrapper = creatWrapper()
+    const wrapper = createWrapper()
     wrapper.unmount()
   })
 
   it('renders static parts', () => {
-    const wrapper = creatWrapper()
+    const wrapper = createWrapper()
```
Comment (JST5000) [inline src/containers/PayStrings/PayStringMappingsTable/test/PayStringMappingsTable.test.js:21]: nice catch :) (Took me a couple reads to see the difference)

File: src/rippled/payString.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export interface PayStringResponse {
+  payString: string
+  addresses: {
+    paymentNetwork: string
+    environment: string
+    addressDetailsType: string
+    addressDetails: {
+      address: string
+      tag: string // optional
```
Comment (JST5000) [inline src/rippled/payString.ts:18]: Should this be written as `tag?: string` instead of this comment? or is the implication that it can be empty?

**Review by JST5000 — APPROVED** [review]
LGTM after a small question

File: src/rippled/payString.ts (hunk 2 above)
Comment (ckniffen) [reply src/rippled/payString.ts]: Just confirmed the output from the server and it will be undefined.

**Pull request merged**